plugins/inputs/multifile/multifile.go

```diff
@@ -0,0 +1,160 @@
+package multifile
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// `toml:"name_override"`
+
+type MultiFile struct {
+	BaseDir   string
+	FailEarly bool
+	Files     []File `toml:"file"`
+	Tags      map[string]string
+
+	initialized bool
+}
+
+type File struct {
+	Name       string `toml:"file"`
+	Dest       string
+	Conversion string
+}
+
+const sampleConfig = `
+  name_override = "sensor"
+  base_dir = "/sys/bus/i2c/devices/1-0076/iio:device0"
+  [inputs.multifile.tags]
+    location = "server_room"
+  [[inputs.multifile.file]]
+    file = "name"
+    dest = "type"
+    conversion = "tag"
+  [[inputs.multifile.file]]
+    file = "in_pressure_input"
+    dest = "pressure"
+    conversion = "float"
+  [[inputs.multifile.file]]
+    file = "in_temp_input"
+    dest = "temperature"
+    conversion = "float(3)"
+  [[inputs.multifile.file]]
+    file = "in_humidityrelative_input"
+    dest = "humidityrelative"
+    conversion = "float(3)"
+`
+
+// SampleConfig returns the default configuration of the Input
+func (m *MultiFile) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *MultiFile) Description() string {
+	return "Aggregates the contents of multiple files into a single point"
+}
+
+func (m *MultiFile) init() {
+	if m.initialized {
+		return
+	}
+
+	for i, file := range m.Files {
+		if m.BaseDir != "" {
+			m.Files[i].Name = path.Join(m.BaseDir, file.Name)
+		}
+		if file.Dest == "" {
+			m.Files[i].Dest = path.Base(file.Name)
+		}
+	}
+
+	m.initialized = true
+}
+
+func (m *MultiFile) Gather(acc telegraf.Accumulator) error {
+	m.init()
+	now := time.Now()
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+
+	for key, value := range m.Tags {
```

Comment:
Tags setting is actually built in on any input plugin, you can remove this code and it will still work.

Reply:
I needed quite a while to get this working, can you tell me if I can also remove [the Tags Array in Line 23]?(https://github.com/influxdata/telegraf/pull/5256/files/44f0acfe259becb9d4db6ee8b35e91a770ad6152#diff-4e89817cac1e51f96268b11e514a37feR23)